src/main/java/seedu/address/model/person/Birthday.java

```diff
@@ -4,45 +4,69 @@
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
 import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
 
 /**
  * Represents a Person's birthday in the address book.
- * Guarantees: immutable; is valid as declared in {@link #isValidBirthday(String)}
+ * Guarantees: immutable; is valid as declared in {@link #isValidFormat(String)}
  */
 public class Birthday {
 
-    public static final String MESSAGE_CONSTRAINTS = "Birthdays should come in the form of yyyy-MM-dd";
-    public static final String VALIDATION_REGEX = "^\\d{4}\\-(0?[1-9]|1[012])\\-(0?[1-9]|[12][0-9]|3[01])$";
-    public final LocalDate birthday;
+    public static final String MESSAGE_CONSTRAINTS = "Birthdays should come in the form of ddMMyyyy";
+    public static final String MESSAGE_INVALID_DATE = "Birthday is not a valid date";
+    public static final String VALIDATION_REGEX = "\\d{8}";
+    private static final DateTimeFormatter BIRTHDATE_FORMATTER = DateTimeFormatter.ofPattern("ddMMyyyy");
+    public final LocalDate birthdate;
 
     /**
      * Constructs an {@code Birthday}.
      *
-     * @param birthday A valid email address.
+     * @param birthdate A valid date value.
      */
-    public Birthday(String birthday) {
-        requireNonNull(birthday);
-        checkArgument(isValidBirthday(birthday), MESSAGE_CONSTRAINTS);
-        this.birthday = LocalDate.parse(birthday);
+    public Birthday(String birthdate) {
+        requireNonNull(birthdate);
+        checkArgument(isValidFormat(birthdate), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidDate(birthdate), MESSAGE_INVALID_DATE);
+        this.birthdate = LocalDate.parse(birthdate, BIRTHDATE_FORMATTER);
     }
 
     /**
-     * Returns if a given string is a valid birthday.
+     * Returns if a given string is in valid format.
      */
-    public static boolean isValidBirthday(String birthday) {
+    public static boolean isValidFormat(String birthday) {
         return birthday.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Returns if a given string is a valid date.
```

Comment:
Would "Checks if a given string is a valid date" be better?

Reply:
Yup agree.

Reply:
Do also add in the `param` and `return` fields